packages/miniflare/src/workers/core/proxy.worker.ts

```diff
@@ -119,7 +125,11 @@ export class ProxyServer implements DurableObject {
 			// should only ever return `Object`, as none override `Symbol.toStringTag`
 			// https://tc39.es/ecma262/multipage/fundamental-objects.html#sec-object.prototype.tostring
 			const type = getType(value);
-			if ((type === "Object" && !isPlainObject(value)) || type === "Promise") {
+			const isInternalClass = isInternal(value);
+			if (
+				((type === "Object" || isInternalClass) && !isPlainObject(value)) ||
+				type === "Promise"
```

Comment:
Out of interest... Would a `Promise` be an internal class?

Reply:
It doesn't seem to be (removing this makes 114 tests fail)

Reply:
Unlikely, I think. The type is derived from [Object.prototype.toString()](https://tc39.es/ecma262/multipage/fundamental-objects.html#sec-object.prototype.tostring) which returns `Promise` only if `Symbol.toStringTag` is set to `Promise` but `Symbol.toStringTag` is missing on all internal classes until the release yesterday.

Reply:
Oh right, I think `Promise` is a built-in V8 class, not an workerd internal class, so it would not be marked "internal".